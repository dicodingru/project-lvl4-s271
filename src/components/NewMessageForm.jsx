import React from 'react';

const NewMessageForm = () => (
  <form>
    <input type="text" />
    <button type="submit">Send</button>
  </form>
);

export default NewMessageForm;
