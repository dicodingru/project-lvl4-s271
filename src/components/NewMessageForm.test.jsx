import React from 'react';
import renderer from 'react-test-renderer';
import NewMessageForm from './NewMessageForm';

describe('NewMessageForm component', () => {
  let component;
  const messages = ['msg1', 'msg2'];

  beforeEach(() => {
    component = mount(<NewMessageForm />);
  });

  test('should render successfully', () => {
    expect(component.exists()).toEqual(true);
  });

  test('should render input field', () => {
    expect(component.find('input').length).toBe(1);
  });

  test('should render submit button', () => {
    expect(component.find('button').length).toBe(1);
  });

  test('should match snapshot', () => {
    component = renderer.create(<NewMessageForm />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
