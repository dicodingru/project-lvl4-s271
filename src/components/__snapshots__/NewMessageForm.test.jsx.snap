// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageForm component should match snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  >
    Send
  </button>
</form>
`;
